lib/ObjWriter/objwriter.h

```diff
@@ -0,0 +1,261 @@
+//===---- objwriter.h --------------------------------*- C++ -*-===//
+//
+// object writer
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/CodeGen/AsmPrinter.h"
+#include "llvm/DebugInfo/CodeView/CodeView.h"
+#include "llvm/DebugInfo/CodeView/Line.h"
+#include "llvm/DebugInfo/CodeView/SymbolRecord.h"
+#include "llvm/MC/MCAsmBackend.h"
+#include "llvm/MC/MCAsmInfo.h"
+#include "llvm/MC/MCContext.h"
+#include "llvm/MC/MCDwarf.h"
+#include "llvm/MC/MCInstPrinter.h"
+#include "llvm/MC/MCInstrInfo.h"
+#include "llvm/MC/MCObjectFileInfo.h"
+#include "llvm/MC/MCParser/AsmLexer.h"
+#include "llvm/MC/MCRegisterInfo.h"
+#include "llvm/MC/MCSectionCOFF.h"
+#include "llvm/MC/MCSectionELF.h"
+#include "llvm/MC/MCSectionMachO.h"
+#include "llvm/MC/MCStreamer.h"
+#include "llvm/MC/MCSubtargetInfo.h"
+#include "llvm/MC/MCParser/MCTargetAsmParser.h"
+#include "llvm/MC/MCTargetOptionsCommandFlags.h"
+#include "llvm/MC/MCWinCOFFStreamer.h"
+#include "llvm/Support/COFF.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Compression.h"
+#include "llvm/Support/FileUtilities.h"
+#include "llvm/Support/FormattedStream.h"
+#include "llvm/Support/Host.h"
+#include "llvm/Support/ManagedStatic.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/PrettyStackTrace.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/TargetRegistry.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "llvm/Support/Win64EH.h"
+#include "llvm/Target/TargetMachine.h"
+#include "llvm/Target/TargetOptions.h"
+#include "llvm/Target/TargetSubtargetInfo.h"
+#include "llvm/Support/ELF.h"
+#include "cfi.h"
+#include <string>
+#include "jitDebugInfo.h"
+
+using namespace llvm;
+using namespace llvm::codeview;
+
+enum CustomSectionAttributes : int32_t
+{
+	CustomSectionAttributes_ReadOnly = 0x0000,
+	CustomSectionAttributes_Writeable = 0x0001,
+	CustomSectionAttributes_Executable = 0x0002,
+	CustomSectionAttributes_MachO_Init_Func_Pointers = 0x0100,
+};
+
+enum class RelocType
+{
+	IMAGE_REL_BASED_ABSOLUTE = 0x00,
+	IMAGE_REL_BASED_HIGHLOW = 0x03,
+	IMAGE_REL_BASED_DIR64 = 0x0A,
+	IMAGE_REL_BASED_REL32 = 0x10,
+};
+
+class ObjectWriter
+{
+public:
+	bool init(StringRef FunctionName);
```

Comment:
We should be consistent with naming here. Either camel / pascal case for all class members.